src/masternodes/govvariables/attributes.cpp

```diff
@@ -60,6 +59,30 @@ static ResVal<CAmount> VerifyPct(const std::string& str) {
     return resVal;
 }
 
+static ResVal<CTokenCurrencyPair> VerifyCurrencyPair(const std::string& str) {
+    const auto value = KeyBreaker(str);
+    if (value.size() != 2) {
+        return Res::Err("Exactly two entires expected for currency pair");
+    }
+    auto token = trim_all_ws(value[0]).substr(0, CToken::MAX_TOKEN_SYMBOL_LENGTH);
+    auto currency = trim_all_ws(value[1]).substr(0, CToken::MAX_TOKEN_SYMBOL_LENGTH);
+    if (token.empty() || currency.empty()) {
+        return Res::Err("Empty token / currency");
+    }
+    return {{token, currency}, Res::Ok()};
+}
+
+static ResVal<std::string> VerifyBool(const std::string& str) {
```

Comment:
just a note, but: "VerifyCurrencyPair" makes a currencyPair Result from the string. Why is VerifyBool returning the string in a result and not returning a bool-result?

Reply:
It cannot be bool as ResVal already overloads bool, conflicts in ResVal are.

```
operator T() const
operator bool() const
```

It is simple enough here to returns the value and compare to "true" when applying to `attribValue`.

Reply:
Right. It can return `int` then just check `== 1`. We could avoid strings compare.

Reply:
Okey dokey. Let's do that then.

Reply:
We always have to compare string as the value is a string, we can do it in VerifyBool, safer than multiple locations.

```
static ResVal<int> VerifyBool(const std::string& str) {
    if (str != "true" && str != "false") {
        return Res::Err(R"(Value must be either "true" or "false")");
    }
    return {str == "true", Res::Ok()};
}
```

Then we set it with...

`attribValue = bool(*res.val);`

Reply:
https://github.com/DeFiCh/ain/pull/1108/commits/a56455f582933bc42797e14c2ea1bb81d7d98266